File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> pMap = new HashMap<String, Integer>();
		for(String str : participant) {
			pMap.put(str, pMap.getOrDefault(str, 0)+1);
		}
		for(String str : completion) {
			pMap.put(str, pMap.getOrDefault(str, 0)-1);
		}
		for(String key : pMap.keySet()) {
			if(pMap.get(key) != 0) {
				answer = key;
			}
		}
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
const solution = (participant, completion) => {
    const participantMap = new Map();
    
    for(const name of participant) {
        const value = participantMap.get(name);
        
        participantMap.set(name, (value ?? 0) + 1);
    }
    
    for(const name of completion) {
        const value = participantMap.get(name);
        
        if(value) {
            participantMap.set(name, value - 1);
        } else {
            return name;
        }
    }
    
    for(const [name, count] of participantMap) {
        if(count !== 0) {
            return name;      
        }
    }
}